src/app/cli/src/init/client.ml

```diff
@@ -1789,90 +1789,82 @@ let add_peers_graphql =
                   } ) ) ) )
 
 let compile_time_constants =
-  let genesis_constants = Genesis_constants.Compiled.genesis_constants in
-  let constraint_constants = Genesis_constants.Compiled.constraint_constants in
-  let proof_level = Genesis_constants.Compiled.proof_level in
+  let open Command.Let_syntax in
   Command.async
     ~summary:"Print a JSON map of the compile-time consensus parameters"
-    (Command.Param.return (fun () ->
+    (let%map_open config_file = Cli_lib.Flag.config_files in
+     fun () ->
+       let home = Core.Sys.home_directory () in
+       let conf_dir = home ^/ Cli_lib.Default.conf_dir_name in
+       let genesis_dir =
          let home = Core.Sys.home_directory () in
-         let conf_dir = home ^/ Cli_lib.Default.conf_dir_name in
-         let genesis_dir =
-           let home = Core.Sys.home_directory () in
-           home ^/ Cli_lib.Default.conf_dir_name
-         in
-         let config_file =
-           match Sys.getenv "MINA_CONFIG_FILE" with
-           | Some config_file ->
-               config_file
-           | None ->
-               conf_dir ^/ "daemon.json"
-         in
-         let open Async in
+         home ^/ Cli_lib.Default.conf_dir_name
+       in
+       let open Deferred.Let_syntax in
+       let%map ({ consensus_constants; _ } as precomputed_values), _ =
+         (* This is kind of ugly because we are allowing for supplying a runtime_config value directly, rather than force what is read from the environment *)
+         (* TODO: See if we can initialize consensus_constants without also initializing the ledger *)
```

Comment:
This is heinous.

Reply:
Well, it was strange (at least) even before, with combination of compile-time constants, `daemon.json` and `MINA_CONFIG_FILE` env variable potentially contributing to the output of the CLI command with misleading _compile-time consensus parameters_ description.

The current modification was meant to standardize the behavior of the command, so that if we're looking up `daemon.json` and `MINA_CONFIG_FILE` env variable, we look up all of the other magic paths. And also allow config file to be specified with a flag.

TL;DR Name and description of the command was misleading long before the recent changes.